pinot-core/src/main/java/com/linkedin/pinot/core/operator/query/MetadataBasedAggregationOperator.java

```diff
@@ -67,13 +71,28 @@ protected IntermediateResultsBlock getNextBlock() {
       AggregationFunction function = aggregationFunctionContext.getAggregationFunction();
       AggregationFunctionFactory.AggregationFunctionType functionType =
           AggregationFunctionFactory.AggregationFunctionType.valueOf(function.getName().toUpperCase());
+      String column = aggregationFunctionContext.getAggregationColumns()[0];
 
-      // TODO: Add support for more aggregation functions that can be served with metadata.
+      SegmentMetadataImpl segmentMetadata = (SegmentMetadataImpl) _segmentMetadata;
       AggregationResultHolder resultHolder;
       switch (functionType) {
         case COUNT:
           resultHolder = new DoubleAggregationResultHolder(totalRawDocs);
           break;
+        case MIN:
+          String minValue = segmentMetadata.getColumnMetadataFor(column).getMinValue().toString();
+          resultHolder = new DoubleAggregationResultHolder(Double.valueOf(minValue));
```

Comment:
what happens if its time column and the format is SDF and cannot be parsed. Let us try to use right result holder based on the column data type.  If this was not supported in the current version, please file another issue to track this. Ideally, we want to MIN/MAX etc on any data type.

Reply:
min max doesn't support non numeric columns right no. Created https://github.com/linkedin/pinot/issues/2208 to track that.